.github/workflows/update-zarf-injector.yaml

```diff
@@ -0,0 +1,88 @@
+name: Zarf Injector Update
+
+on:
+  pull_request:
```

Comment:
Because this repo is public, I think that adding an on pull request workflow will allow any fork to run arbitrary code (unless we have stringent repo/branch settings).

We may want the update injector workflow to work similar to the publish workflow, `on push to main` 
- check if it needs updating, then update else exit 

cc: @bburky

Reply:
I removed the additional logic, the original intent was to have it work alongside renovate prior to publishing.